src/main/java/com/revlearn/team1/service/transaction/StripeService.java

```diff
@@ -0,0 +1,83 @@
+package com.revlearn.team1.service.transaction;
+
+import com.revlearn.team1.dto.transaction.TransactionResponseDTO;
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import com.stripe.model.checkout.Session;
+import com.stripe.param.checkout.SessionCreateParams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.stereotype.Service;
+
+import com.revlearn.team1.dto.transaction.TransactionRequestDTO;
+import com.revlearn.team1.mapper.TransactionMapper;
+import com.revlearn.team1.model.TransactionModel;
+import com.revlearn.team1.repository.TransactionRepo;
+
+@Service
+public class StripeService {
+    @Autowired
+    TransactionRepo transactionRepo; // For future use with Kafka
+
+    @Autowired
+    TransactionMapper transactionMapper; // For future use with Kafka
+
+    @Autowired
+    KafkaTemplate<String, Session> kafkaTemplate;
+
+    @Value("${STRIPE_API_KEY}")
+    private String stripeApiKey;
+
+    @Value("${CLIENT_URL}")
+    private String clientUrl;
+
+    public TransactionResponseDTO checkout(TransactionRequestDTO request) throws StripeException {
+        // Set the Stripe API key
```

Comment:
i didn't catch this last time, but can you wrap this function in a try-catch block